components/proxy/src/main/java/com/hotels/styx/StyxServer.java

```diff
@@ -220,8 +265,14 @@ private static StartupConfig parseStartupConfig(String[] args) {
     @Override
     protected void doStart() {
         printBanner();
-        this.serviceManager.addListener(new ServerStartListener(this));
-        this.serviceManager.startAsync();
+        CompletableFuture.runAsync(() -> {
+            // doStart should return quicly. Therefore offload waiting on a separate thread:
+            this.phase1Services.addListener(new Phase1ServerStartListener(this));
+            this.phase1Services.startAsync().awaitHealthy();
+
+            this.phase2Services.addListener(new ServerStartListener(this));
+            this.phase2Services.startAsync();
+        });
```

Comment:
If anything in here fails, is the exception being logged?

Reply:
I have made some changes to log them from StyxServer logger.

Sadly the `shutdownLogging` mechanism (see in StyxServer.java) prevents the tests from working correctly. I need to look into removing this in near future.